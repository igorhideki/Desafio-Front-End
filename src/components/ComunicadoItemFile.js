import React from 'react';
import PropTypes from 'prop-types';
import api from '../requests/apis';

const handleAnexoTypeIcon = (type) => {
  const isImage = /.png$|.jpg$|.jpeg$|.gif$/;
  const isPDF = /.pdf$/;
  if (isImage.test(type)) {
    return 'far fa-file-image';
  }
  if (isPDF.test(type)) {
    return 'far fa-file-pdf';
  }
  return 'far fa-file';
};

const ComunicadoItemFile = (props) => (
  <div className="anexo">
    <span className="anexo__icon">
      <i className={handleAnexoTypeIcon(props.url)}></i>
    </span>
    <span className="anexo__name">{props.name}</span>
    <span
      className="anexo__link"
      onClick={() => api.downloadFile(props.url)}
    >
      <i className="fas fa-cloud-download-alt"></i>
    </span>
  </div>
);

ComunicadoItemFile.propTypes = {
  name: PropTypes.string,
  url: PropTypes.string
};

export default ComunicadoItemFile;
