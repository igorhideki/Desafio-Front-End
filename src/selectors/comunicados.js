import moment from 'moment';

const getVisiblesComunicados = (comunicados, { type = -1 }) => {
  const formatDate = 'YYYY-MM-DD HH:mm';
  return comunicados.filter((comunicado) => {
    if (type === -1) {
      return comunicado;
    }
    return comunicado.type === type;
  }).sort((a, b) => {
    return moment(a.date, formatDate).format() < moment(b.date, formatDate).format() ? 1 : -1;
  });
};

export default getVisiblesComunicados;

