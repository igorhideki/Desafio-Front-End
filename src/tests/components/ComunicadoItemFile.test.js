import React from 'react';
import { shallow } from 'enzyme';
import ComunicadoItemFile from '../../components/ComunicadoItemFile';
import comunicados from '../fixtures/comunicados';

test('should render ComunicadoItemFile correctly', () => {
  const wrapper = shallow(<ComunicadoItemFile key={comunicados[2].files[0].name} {...comunicados[2].files[0]} />);
  expect(wrapper).toMatchSnapshot();
});
