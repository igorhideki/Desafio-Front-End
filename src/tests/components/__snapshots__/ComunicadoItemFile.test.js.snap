// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ComunicadoItemFile correctly 1`] = `
<div
  className="anexo"
>
  <span
    className="anexo__icon"
  >
    <i
      className="far fa-file-image"
    />
  </span>
  <span
    className="anexo__name"
  >
    Material de divulgação
  </span>
  <span
    className="anexo__link"
    onClick={[Function]}
  >
    <i
      className="fas fa-cloud-download-alt"
    />
  </span>
</div>
`;
